lib/yang_wrappers.c

```diff
@@ -792,6 +792,30 @@ struct yang_data *yang_data_new_empty(const char *xpath)
 	return yang_data_new(xpath, NULL);
 }
 
+bool yang_dnode_get_empty(const struct lyd_node *dnode, const char *xpath_fmt,
+			  ...)
+{
+	const struct lyd_node_leaf_list *dleaf;
+
+	assert(dnode);
+	if (xpath_fmt) {
+		va_list ap;
+		char xpath[XPATH_MAXLEN];
+
+		va_start(ap, xpath_fmt);
+		vsnprintf(xpath, sizeof(xpath), xpath_fmt, ap);
+		va_end(ap);
+		dnode = yang_dnode_get(dnode, xpath);
+		YANG_DNODE_GET_ASSERT(dnode, xpath);
```

Comment:
The point of empty leafs is that they can exist or not, so this assert here doesn't make sense.

Also, empty leafs can't have a default value and because of that we should avoid using them for configuration data, otherwise "show configuration ... with-defaults" won't work as expected. Boolean leafs should be preferred instead. That said, I think it's useful to have this wrapper just in case it happens to be necessary sometime in the future.

Reply:
@rwestphal  assert is removed. 
There are config models in zebra uses type empty. [PR 6758 ](https://github.com/FRRouting/frr/pull/6758) converted one of the data model from empty type to boolean leaf. 
This needs to be documented to avoid using empty type for config data model. 

Though I agree that this wrapper can be handy to keep.

Reply:
Maybe this function should be a wrapper to `yang_dnode_exists` (plus type check)? I also don't see a point to crash if it doesn't exist...

For booleans it make sense because there it is possible to have three states:
- default (non existent or not configured)
- false
- true

The main difference between this and `boolean` is that this type has only 2 options: it exists or it doesn't. So why would you make this function crash if we are only testing it for existence? Or maybe do we even need this?